.github/workflows/functional-test.yaml

```diff
@@ -715,10 +715,14 @@ jobs:
             for (const run of response.data.workflow_runs) {
               if (run.conclusion === 'failure') {
                 failureCount++;
-              } else {
+              } else if (run.conclusion === 'success') {
```

Comment:
I am trying to understand better. So, `listWorkflowRuns` above will bring 10 workflows. But I think what we should do is that we should take a look at the **last completed workflow outcome** and, if that is **failure**, then we can create an issue.

If we check the last 10 all the time, we may end up creating multiple issues for the same failures.  I think adding the completed status check to the call above and setting `per_page` to 1 would be sufficient. If the last completed workflow has failed, then just set the **failureCount** to 2 and create the issue.

Suggestions:
- We can add a status **COMPLETED** check to the GitHub call above. We can see different statuses [here](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository).

Reply:
Please see this screen shot. When I cancelled workflow, cancelled run was also in completed status. 
<img width="1169" alt="image" src="https://github.com/radius-project/radius/assets/11863108/b4401d63-2d6a-487f-9b17-adb4955617f4">

Because we want to treat only `failure` conclusion as a failed run, we need to get 1+ actions runs and then linear search to find the recent failure in the list. So I was fetching 10 action runs and then scan the workflow status to find the failure and ignore the other statuses, such as cancelled.

Also, I want to allow us to adjust the acceptable failures. that's why I add `ISSUE_CREATE_THRESHOLD` variable. I can add more comments for the clarification.

Reply:
I created new composite action to be shared. Please review the PR again.